packages/utah-design-system/src/components/Sherlock.jsx

```diff
@@ -204,16 +204,147 @@ export const masqueradeProvider = (url, wkid) => {
   };
 };
 
+export const featureServiceProvider = (
+  url,
+  searchField,
+  contextField = null,
+) => {
+  let oidField;
+  const init = async (signal) => {
+    // get oid field name, and validate searchField and contextField
+    const serviceResponse = await fetch(`${url}?f=json`, { signal });
+    const serviceJson = await serviceResponse.json();
+
+    let searchFieldValidated = false;
+    let contextFieldValidated = false;
+    for (const field of serviceJson.fields) {
+      if (field.type === 'esriFieldTypeOID') {
+        oidField = field.name;
+      }
+
+      if (field.name === searchField) {
+        searchFieldValidated = true;
+      }
+
+      if (contextField && field.name === contextField) {
+        contextFieldValidated = true;
+      }
+    }
+
+    if (!oidField) {
+      throw new Error('OID field not found');
+    }
+
+    const fieldsList = serviceJson.fields.map((f) => f.name).join(', ');
+    if (!searchFieldValidated) {
+      throw new Error(
+        `Field: ${searchField} not found in service fields: ${fieldsList}`,
+      );
+    }
+
+    if (contextField && !contextFieldValidated) {
+      throw new Error(
+        `Field: ${contextField} not found in service fields: ${fieldsList}`,
+      );
+    }
+  };
+
+  return {
+    load: async ({ signal, filterText, maxResults = 10 }) => {
+      if (!oidField) {
+        await init(signal);
+      }
+
+      if ((filterText?.length ?? 0) < 3) {
+        return { items: [] };
+      }
+
+      const searchParams = new URLSearchParams({
+        f: 'json',
+        where: `UPPER(${searchField}) LIKE UPPER('%${filterText}%')`,
+        outFields: [oidField, searchField, contextField].join(','),
+        returnGeometry: false,
+        resultRecordCount: maxResults,
+      });
+
+      const response = await fetch(`${url}/query?${searchParams.toString()}`, {
+        signal,
+      });
+      const responseJson = await response.json();
+
+      const suggestions = responseJson.features.map((feature) => {
+        return {
+          name: feature.attributes[searchField],
+          context: contextField ? feature.attributes[contextField] : null,
+          key: feature.attributes[oidField],
+        };
+      });
```

Comment:
**suggestion**: this could be more defensive. This assumes a lot of the things that can go wrong here won't. 

For instance, the fetch can return a not ok response or time out. 
The reponse.json can be empty or have the arcgis error format or be totally broken in the case of a 500 response.

Reply:
disregard this. the hook manages loading and error states